processor/src/test/java/com/joom/lightsaber/processor/end2end/End2EndTest.kt

```diff
@@ -0,0 +1,126 @@
+package com.joom.lightsaber.processor.end2end
+
+import com.joom.lightsaber.processor.ErrorReporter
+import com.joom.lightsaber.processor.ErrorReporterImpl
+import com.joom.lightsaber.processor.LightsaberParameters
+import com.joom.lightsaber.processor.LightsaberProcessor
+import java.io.File
+import java.net.URI
+import java.net.URL
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import kotlin.io.path.absolutePathString
+import kotlin.io.path.exists
+import kotlin.io.path.extension
+import kotlin.io.path.isDirectory
+import kotlin.streams.toList
+import org.jetbrains.kotlin.cli.common.ExitCode
+import org.jetbrains.kotlin.cli.jvm.K2JVMCompiler
+
+open class End2EndTest(private val root: Path) {
```

Comment:
I believe this class should be a `TestRule` and should be injected into tests using `@Rule`. And I suggest calling it something like `IntegrationTestRule`.

Reply:
done